services/spar/src/Spar/Data.hs

```diff
@@ -130,16 +130,12 @@ insertUser (SAML.UserRef tenant subject) uid = retry x5 . write ins $ params Quo
     ins = "INSERT INTO user (idp, sso_id, uid) VALUES (?, ?, ?)"
 
 getUser :: (HasCallStack, MonadClient m) => SAML.UserRef -> m (Maybe UserId)
-getUser (SAML.UserRef tenant subject) = (retry x1 . query1 sel $ params Quorum (tenant', subject')) <&> \case
+getUser (SAML.UserRef tenant subject) = (retry x1 . query1 sel $ params Quorum (tenant, subject)) <&> \case
   Just (Identity (Just (UUID.fromText -> Just uuid))) -> Just $ Id uuid
   _ -> Nothing
   where
-    tenant', subject' :: ST
-    tenant'  = cs $ SAML.encodeElem tenant
-    subject' = cs $ SAML.encodeElem subject
-
-    sel :: PrepQuery R (ST, ST) (Identity (Maybe ST))
-    sel = "SELECT uid FROM authresp WHERE idp = ? AND sso_id = ?"
```

Comment:
`UserId` (or any `Id` type) has CQL instance so it's better to use that instead of using `UUID.fromText`. Also, I think `getUser` should look like this:

```
getUser :: (HasCallStack, MonadClient m) => SAML.UserRef -> m (Maybe UserId)
getUser (SAML.UserRef tenant subject) = fmap runIdentity <$>
    retry x1 (query1 sel $ params Quorum (tenant, subject))
  where
    sel :: PrepQuery R (SAML.Issuer, SAML.NameID) (Identity UserId)
    sel = "SELECT uid FROM user WHERE issuer = ? AND sso_id = ?"
```

Note that the return type of `sel` should be `Identity UserId`; i.e., given the schema, a `user_id` must always be set for a pair of `issuer, sso_id`. Only if a field is optional should the query return a `Maybe`. The `query1` function will then return a `Maybe`, given that the primary was not found.

Reply:
yes to all your comments.  6dc86a35f9a779a751fc7fef7ddb41becb5a06cf

thanks!